lib/ansible/plugins/connection/winrm.py

```diff
@@ -49,14 +54,14 @@
 from ansible.utils.hashing import secure_hash
 from ansible.utils.path import makedirs_safe
 from ansible.utils.unicode import to_bytes, to_unicode, to_str
-from ansible.utils.vars import combine_vars
 
 try:
     from __main__ import display
 except ImportError:
     from ansible.utils.display import Display
     display = Display()
 
+HALF_MB = 2**19  # 0.5MB chunks
```

Comment:
I think I may just undo that change and revisit. There are two different hardcoded buffersizes in that module

``` python

      # FUTURE: determine buffer size at runtime via remote winrm config?
      def _put_file_stdin_iterator(self, in_path, out_path, buffer_size=250000):
          in_size = os.path.getsize(to_bytes(in_path, errors='strict'))
          offset = 0
```
and the one changed in this patch

``` python
    def fetch_file(self, in_path, out_path):
          super(Connection, self).fetch_file(in_path, out_path)
          in_path = self._shell._unquote(in_path)
          out_path = out_path.replace('\\', '/')
          display.vvv('FETCH "%s" TO "%s"' % (in_path, out_path), host=self._winrm_host)
          buffer_size = HALF_MB
```

Maybe 
``` python
FETCHFILE_BUFSIZE = 2**19  # in bytes, 0.5MB
PUTFILE_BUFSIZE = 250000
```
(PUTFILE_STDIN_BUFSIZE? )